conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandlerTest.java

```diff
@@ -80,6 +86,74 @@ public void handlesServiceException() throws IOException {
                 .contains("\"parameters\":{\"foo\":\"bar\"}}");
     }
 
+    @Test
+    public void checkServiceExceptionArgumentsSerializedWithToString() throws IOException {
+        record Argument(
+                String field1, Optional<Integer> maybeField2, List<String> collectionField, String nullString) {}
+        exception = new ServiceException(
+                ErrorType.CONFLICT,
+                SafeArg.of("arg", new Argument("foo", Optional.of(42), List.of("bar", "baz"), null)));
+        HttpURLConnection connection = execute();
+
+        assertThat(connection.getResponseCode()).isEqualTo(ErrorType.CONFLICT.httpErrorCode());
+        String expectedSerializedArg =
+                "\"Argument[field1=foo, maybeField2=Optional[42], collectionField=[bar, baz], nullString=null]\"";
+        assertThat(getErrorBody(connection))
```

Comment:
We could deserialize into the `ConjureError` type here in order to get structured data -- that way tests won't fail when order of args changes, as order is not guaranteed by conjure wire spec

Reply:
Updated to deserialize into a `ConjureError` with `ObjectMappers.newClientJsonMapper()`.